src/hooks/useENSAvatar.ts

```diff
@@ -134,11 +136,15 @@ function useERC1155Uri(
   const contract = useERC1155Contract(contractAddress)
   const balance = useSingleCallResult(contract, 'balanceOf', accountArgument)
   const uri = useSingleCallResult(contract, 'uri', idArgument)
+  /* ERC-1155 allows a generic {id} in the URL, so prepare to replace if relevant,
+     in lowercase hexadecimal (with no 0x prefix) and leading zero padded to 64 hex characters.
+  */
+  const idHex = id ? hexZeroPad(BigNumber.from(id).toHexString(), 32).substring(2) : id
   return useMemo(
     () => ({
-      uri: !enforceOwnership || balance.result?.[0] > 0 ? uri.result?.[0] : undefined,
+      uri: !enforceOwnership || balance.result?.[0] > 0 ? uri.result?.[0].replace('{id}', idHex) : undefined,
```

Comment:
this should use `replaceAll`
e.g. `'{id}{id}'.replace('{id}','test')` returns `'test{id}'`
also think you have to use the optional for it, `?.replaceAll`

Reply:
Ok.